rules/nette/src/Rector/Class_/ExtractTemplateClassForPresenterRector.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Nette\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://blog.nette.org/en/latte-how-to-use-type-system#toc-how-to-declare-types
+ *
+ * @see \Rector\Nette\Tests\Rector\Class_\ExtractTemplateClassForPresenterRector\ExtractTemplateClassForPresenterRectorTest
+ */
+final class ExtractTemplateClassForPresenterRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Decouples template class for presenter', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Nette\Application\UI\Presenter;
+
+final class SomePresenter extends Presenter
+{
+    public function render()
+    {
+    }
+}
+
+// some_template.latte
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+use Nette\Application\UI\Presenter;
+
+/**
+ * @property SomeTemplate $template
```

Comment:
would it be possible also to create this class with all properties?

Reply:
Yes. I think the properties should be collected from the related template.

Is there a some convention algorithm to find template based on presenter + method? That would be useful

Reply:
Convention (from nette/application) is to find class which extends `Nette\Bridges\ApplicationLatte\Template` in same namespace as Presenter with the same name as Presenter but word "Presenter" is replaced to "Template". You implemented it well.

For presenter + method there is no convention. I send PR, we will see :)

Reply:
I saw this but it seemed rather weird. Now we have to add `Template` to magic place + mention the annotation in the class. Otherwise one or the other won't work.

Reply:
I like this approach of yours:

```php
        /** @var SomeDefaultTemplate $template */
        $template = $this->createTemplate(SomeDefaultTemplate::class);
```

Explicit, and clear.

Reply:
Well, yes it is explicit and clear. And also it is the same for all presenters one-action and multi-actions. And I like one other thing with this approach - you can have your Templates in another namespace like Presenters. It is not possible for now with the magic template finding.

Reply:
Feature was merged  https://github.com/nette/application/pull/277, but it changes nothing for Rector :)

Reply:
Maybe one thing. You can use $this->template now instead of $this->createTemplate()

Reply:
createTemplate() will not work when you will have actionSomething and renderSomething and in both you will need the template - https://github.com/nette/application/issues/282